Disco lights to the Station

Evige opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
New type or tube light. It changes color randomly from predefined colors. It kinda works like TubeLight mostly, just added a new type DiscoTube. Also added possibility to add these lights into LightReplacer to change these lights to empty wall light sockets. Disco time!

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
This is just for adding more fun for example to bar - making disco with musician or jukebox playing and players dancing. Brings a little more feeling into the "dancing". The light can sometimes be a little wider than a normal tube light, so sometimes it might give just a little advantage in that sense.

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
Using frameTime as calculating in game override Update function to make the lights work. I tried to optimize it so it doesn't do anything unnecessary calculations if there are no Disco Lights. Added some into .yml files to separate it from the original light tube.

Currently there is no way to get these Disco Lights to the station, but one step at time. Maybe can made with Lathe? Cargo can order? I am up for ideas about that one.

Edit 1): Now available to be ordered from Cargo. One steel typed crate (very similar to janitors lights crate). Contains one disco light box which contains 4 disco lights. Costs 2000.

## Media
Didn't add any new sprites. I used existing ones, just adjusted their colors. Of course I needed to add new data into the .yml files to separate the new DiscoLight from the original light tube.

Edit 2): Updated video
https://github.com/user-attachments/assets/7b4120a2-3ab6-4e82-bde4-f69dd1f3bb4c

<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.
-->

## Requirements
<!-- 
Due to influx of PR's we require to ensure that PR's are following the correct guidelines.

Please take a moment to read these if its your first time.

Check the boxes below to confirm that you have in fact seen these (put an X in the brackets, like [X]):
-->
- [x ] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
I think nothing breaks anything.
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->
:cl:
- add: Added Disco Lights!
<!--
Make sure to take this Changelog template out of the comment block in order for it to show up. Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog.
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->

deathride58 commented:
These are *really* bad for players with epilepsy. You should probably make a cvar that disables the effect outright.

Evige commented:
> These are _really_ bad for players with epilepsy. You should probably make a cvar that disables the effect outright.

That might be true if there are multiple Disco Lights in the same room, or even one might trigger it. Hard to say, but I have kept in mind this possibility. I was thinking if this might be considered to be in the game I have to make somekind of a stripped version of the effect or just disable it like you mentioned.

Lyroth001 commented:
I think these are really cool, but agree there should be some kind of accessibility setting to disable them

UbaserB commented:
The colour change is too fast, and even though I don't have epilepsy or anything similar to that the rapidness of color change still hurts my eyes a bit.

Hreno commented:
I think it'd be better if they were spinning, like emergency lights.

Probably it would make players with epilepsy be worse but I'm clueless about this topic.

Evige commented:
Disco lights have been updated a lot from the start. Thanks for all the commentors. I have updated the lights to rotate like emergency lights and slowed down the color change. Lights can now also be ordered from Cargo. I have updated the description of this PR as well. The updated video of the lights:

https://github.com/user-attachments/assets/df55a780-2b55-4b62-96be-8a4ff85b79d8

MilonPL commented:
Thank you for this contribution! After a bit of discussion with other maintainers, we've decided to close this PR, as this implementation doesn't meet our code standards. If you or anyone else wants to pick this up, it should just modify the existing RGBLightController to work with light bulbs rather than hardcoding RGB logic into PoweredLights.
